topics/go/algorithms/searches/binarysearch/binarysearch.go

```diff
@@ -45,34 +45,35 @@ func binarySearchIterative(sortedList []int, target int) (int, error) {
 // binarySearchRecursive takes the list of the sorted numbers and check it
 // with `recursive` process to find the value and return the index of array or
 // return the error if the value not found.
-func binarySearchRecursive(List []int, Target int, low int, high int) (int, error) {
+func binarySearchRecursive(sortedList []int, target int, leftIdx int, rightIdx int) (int, error) {
 
-	// 	find the middle index of array
-	mid := (low + high) / 2
+	// Calculate the middle index of the list.
+	midIdx := (leftIdx + rightIdx) / 2
 
-	// 	If low is greater than high it means the value not found,
-	//	I returned -1 and an error to show user value not found
-	for low > high {
-		return -1, fmt.Errorf("sorry value not found")
-	}
+	switch {
 
-	// 	check is List[mid] value equal with Target value than return the middle value
-	if List[mid] == Target {
-		return mid, nil
-	}
+	// check if the left index is greater than right index,
+	// it's means the target is not exist.
+	case leftIdx > rightIdx:
+		return -1, fmt.Errorf("target not found")
 
-	// 	If List[mid] is bigger than Target
-	//	I returned the binarySearchRecursive function with changing the high value to mid minus by one
-	if List[mid] > Target {
-		return binarySearchRecursive(List, Target, low, mid-1)
-	}
+	// Check if we found the target.
+	case sortedList[midIdx] == target:
```

Comment:
Start with the assignment like above.

Reply:
I used `VScode`, sometimes `GoLand` or `VIM`. 

I configured `gofmt` and `staticcheck` for all of them.

I checked it directly.

gofmt -w .\binarysearch.go
gofmt -w .\binarysearch_test.go

I guess now it's work

Reply:
I changed the comments for the recursive part as well.